django/db/models/sql/compiler.py

```diff
@@ -830,9 +830,12 @@ def execute_sql(self, result_type=MULTI, chunked_fetch=False):
         try:
             cursor.execute(sql, params)
         except Exception:
-            with self.connection.wrap_database_errors:
-                # Closing a server-side cursor could yield an error
+            try:
+                # Might fail for server-side cursors (eg: connection closed)
                 cursor.close()
+            except Exception:
+                # Ignore clean-up errors. Raise the original error instead.
+                pass
```

Comment:
IMHO the original exception is more interesting than the exception raised by the cleanup code. Can you see a case where the exception raised on `close()` should prevail?